R/ocjr-SQ780.R

```diff
@@ -0,0 +1,607 @@
+  theme(axis.text.x = element_text(angle = 45, vjust=0.5),
+        text=element_text(size=12)) +
+  scale_y_continuous(labels = scales::comma) +
+  labs(title = "Simple Possession Drug Convictions, OSCN Counties",
+       color = "Case Type",
+       subtitle = "Impact of SQ780 Passing",
+       x = "Year",
+       y = "Number of Convictions",
+       caption = "Data represents the total number of parties convicted, \nnot the number of cases filed.") +
+  scale_fill_manual(values=c("#999999", "#E69F00"),
+                    name="Case Type",
+                    labels=c("Felony", "Misdemeanor")) +
+  annotate("text", x = 2018.2, y = 8200, label = "SQ780 in Effect",
+           family = "serif", fontface = "italic", colour = "darkred",
+           size = 3)
+
+parties_2013_2022 |>
+  pivot_wider(names_from = case_type, values_from = n) |>
+  mutate(
+    total = CM + CF,
+    statewide = round(total * 2.3778)
+  ) |>
+  write_csv(here("data/parties_convictions_2013_2022.csv"))
+
+# * Total number of SQ780 offenses charged in 2013-2022 statewide
+
+## Brancen was able to gather a more robust estimate of 780 cases using some of the methodology below + methodology used in the Loft report to extrapolate OSCN estimates to a state-wide estimate.
+clean_df <- read_csv(here("data/clean_df.csv"))
+parties <- read_csv(here("data/parties_2001_2023.csv"))
+
+# One with the requirement that all charges in the case be 780 charges
+all_780_csv <- read_csv(here("data/all_780_summary.csv"))
+
+# At least one charge in the case being a 780 charge
+any_780_csv <- read_csv(here("data/any_780_summary.csv"))
+
+# Will proceed to work with "at least one" SQ780 charge in the case
+
+clean_df |>
+  summarise(
+    n_cases = n_distinct(id),
+    n_counts = n_distinct(id_count),
+    .by = c("fiscal_year", "case_type")
+  ) |>
+  arrange(
+    fiscal_year,
+    case_type
+  ) |>
+  mutate(
+    est_state_cases = n_cases * 2.3778
+  )
+
+any_780_df <- clean_df |>
+  group_by(id) |>
+  mutate(
+    any_780 = any(sq780_drugs | sq780_props)
+  ) |>
+  add_count(any_780) |>
+  ungroup() |>
+  filter(any_780)
+
+any_780_df_out <- any_780_df |>
+  filter(fiscal_year > 2012,
+         fiscal_year < 2023) |>
+  summarise(
+    n_cases = n_distinct(id),
+    n_counts = n_distinct(id_count),
+    .by = c("fiscal_year", "case_type")
+  ) |>
+  arrange(fiscal_year)
+
+convicted_parties_any_780 <- any_780_df |>
+  mutate(
+    conviction = case_when(
+      str_detect(disposition, "CONV") ~ T,
+      str_detect(disposition, "Guilty Plea") ~ T,
+      str_detect(disposition_detail, "(?i)Guilty Plea") ~ T,
+      str_detect(disposition, "DISMIS") ~ F,
+      str_detect(disposition, "(?i)DEFER") ~ F,
+      str_detect(disposition, "ACQUIT") ~ F,
+      str_detect(disposition, "Pending") ~ F,
+      str_detect(disposition, "(?i)TRANSFER") ~ F,
+      str_detect(disposition_detail, "Dismiss") ~ F,
+      TRUE ~ F
+    )
+  ) |>
+  #filter(conviction) |>
```

Comment:
This should be uncommented. Though the `parties` data frame should only include convicted parties, it is reasonable to filter `any_780_df` just in case and for clarity